### PR TITLE
Modify data process to include confidence interval

### PR DESCRIPTION
**Pull request by SaranjeetKaur**

### CODE REVIEW

File: R/data-process.R [hunk 1]
```diff
@@ -1,229 +1,273 @@
...
+#' This function performs data smoothing.
+#'
+#' @param data_weighted_burden A data frame containing weighted values for burden of disease.
+#' @return A data frame with spline smoothing applied for burden of disease.
+#' @export
+gen_data_weighted_burden_spline <- function(data_weighted_burden) {
+
+  ## Only keep those 0 or negative values
+
+  ## Notes: Delete years 27,30,32-33 for ps3-low; Delete years 2028 for ps4-low
```
Comment (jzhu20) [inline R/data-process.R:236]: Can I amend this note to 
## Notes for India project: Delete years 27,30,32-33 for ps3-low; Delete years 2028 for ps4-low?

File: R/data-process.R [hunk 2]
```diff
@@ -1,229 +1,273 @@
...
+
+  data_weighted_burden_mean <- data_weighted_burden |>
+    dplyr::filter(data_weighted_burden$cumdiff_daly_mean <= 0)
+
+  data_weighted_burden_min <- data_weighted_burden |>
+    dplyr::filter(data_weighted_burden$cumdiff_daly_min <= 0)
+
+  ## Notes: Delete years 29, 31 for ps2-high; Delete 37-38 for ps3-low; Delete 33-34 for ps4-middle; Delete 36-38 for ps4-low
```
Comment (jzhu20) [inline R/data-process.R:244]: Also amend to "Notes for India project: Delete years 29, 31 for ps2-high; Delete 37-38 for ps3-low; Delete 33-34 for ps4-middle; Delete 36-38 for ps4-low"

File: R/data-process.R (hunk 1 above)
Comment (SaranjeetKaur) [reply R/data-process.R:236]: Hi @jzhu20, yes I have made the change to this note

File: R/data-process.R (hunk 2 above)
Comment (SaranjeetKaur) [reply R/data-process.R:244]: This also

**Comment by SaranjeetKaur** [comment]
@jzhu20 - I've addressed your feedback and made the required changes.

File: R/data-process.R [hunk 3]
```diff
@@ -1,229 +1,276 @@
...
+  data_weighted_burden_spline$cumdiff_daly_mean <- stats::predict(spline_fit, data_weighted_burden_spline$time)$y
+
+  spline_fit_min <- splines::interpSpline(data_weighted_burden_min$time, data_weighted_burden_min$cumdiff_daly_min)
+  data_weighted_burden_spline$cumdiff_daly_min <- stats::predict(spline_fit_min, data_weighted_burden_spline$time)$y
+
+  ## Use smooth.spline for ps4-low
+  spline_fit_max <- splines::interpSpline(data_weighted_burden_max$time, data_weighted_burden_max$cumdiff_daly_max)
+  data_weighted_burden_spline$cumdiff_daly_max <- stats::predict(spline_fit_max, data_weighted_burden_spline$time)$y
+
+  ## Keep 0 values in the first two years
```
Comment (jzhu20) [inline R/data-process.R:268]: Can we clarify this note further?
"Keep 0 values in the first two years, before policy implementation"

Comment (SaranjeetKaur) [reply R/data-process.R]: @jzhu20 - done!

**Comment by SaranjeetKaur** [comment]
@jzhu20 - since you have already reviewed this PR a couple of times, I am taking the liberty to merge it.

File: R/data-process.R [hunk 4]
```diff
@@ -1,3 +1,60 @@
...
+#'    ```r
+#'    data_weighted_ds_wide_collapse <- gen_data_weighted_ds(data_weighted)
+#'    ```
+#'
+#'5. **`gen_data_weighted_burden`**: Calculates the differences between intervention and baseline values for burden of disease.
+#'    ```r
+#'    data_weighted_burden_wide_collapse <- gen_data_weighted_burden(data_weighted)
+#'    ```
+#'
+#'6. **`gen_data_weighted_burden_spline`**: Performs data smoothing.
```
Comment (jzhu20) [inline R/data-process.R:33]: Performs data smoothing for burden of disease, when necessary. For instance, with only a few simulations, there can be positive values in difference in burden of disease.

File: R/data-process.R [hunk 5]
```diff
@@ -1,229 +1,333 @@
...
+#'    ```r
+#'    data <-  readRDS("data.rds")
+#'    ```
+#'
+#' 2. **`gen_data_mean`**: Calculates weighted mean values for various metrics over years.
+#'    ```r
+#'    data_weighted <- gen_data_mean(data)
+#'    ```
+#'
+#' 3. **`gen_data_weighted_rf`**: Calculates the differences between intervention and baseline values for various metrics.
```
Comment (jzhu20) [inline R/data-process.R:18]: Calculates the differences between intervention and baseline values for **risk factors**.

**Pull request merged**